src/hashrate.rs

```diff
@@ -1,74 +1,25 @@
-use std::time::{Duration, Instant};
-use colored::*;
-
-
-pub fn get_unit(hashrate: f32) -> (f32, String) {
-    let mut hashrate = hashrate as f32;
-    let mut unit = "";
-    let units = vec!["k", "M", "G", "T", "P", "E", "Z", "Y"];
-    for u in units.iter() {
-        if hashrate > 1000.0 {
-            hashrate /= 1000.0;
-            unit = u;
-        } else {
-            break;
-        }
-    }
-    (hashrate, format!("{}H/s", unit))
-}
-
-#[derive(Clone, Debug, PartialEq)]
-pub struct Hashrate {
-    pub worker_id: u32,
-    value: f32,
-    counter: u32,
-    counter_max: u32,
-    sampling_start: Instant,
-    report_start: Instant,
-    report_interval: Duration,
-}
+#[derive(Clone, Copy, Default)]
+pub struct Hashrate(pub f32);
```

Comment:
can you point me a resource about this kind of definition? I first time see this. This is very clean! loved it!

Reply:
OK, I found out. `Tuple struct`. Thanks :)